diff --git a/iceoryx_posh/include/iceoryx_posh/internal/runtime/posh_runtime_impl.hpp b/iceoryx_posh/include/iceoryx_posh/internal/runtime/posh_runtime_impl.hpp
index fb94bdc14..76e29dd89 100644
--- a/iceoryx_posh/include/iceoryx_posh/internal/runtime/posh_runtime_impl.hpp
+++ b/iceoryx_posh/include/iceoryx_posh/internal/runtime/posh_runtime_impl.hpp
@@ -85,7 +85,7 @@ class PoshRuntimeImpl : public PoshRuntime
     /// @copydoc PoshRuntime::sendRequestToRouDi
     bool sendRequestToRouDi(const IpcMessage& msg, IpcMessage& answer) noexcept override;
 
-  protected:
+  public:
     friend class PoshRuntime;
     friend class roudi_env::RuntimeTestInterface;
     friend class posh::experimental::Node;
diff --git a/iceoryx_posh/include/iceoryx_posh/runtime/posh_runtime.hpp b/iceoryx_posh/include/iceoryx_posh/runtime/posh_runtime.hpp
index 9cc94e067..0cff9ef56 100644
--- a/iceoryx_posh/include/iceoryx_posh/runtime/posh_runtime.hpp
+++ b/iceoryx_posh/include/iceoryx_posh/runtime/posh_runtime.hpp
@@ -165,11 +165,13 @@ class PoshRuntime
     /// @return current runtime factory
     static factory_t& getRuntimeFactory() noexcept;
 
+  public:
     /// @brief sets runtime factory, terminates if given factory is empty
     ///
     /// @param[in] factory function pointer to which the runtime factory should be set
     static void setRuntimeFactory(const factory_t& factory) noexcept;
 
+  protected:
     /// @brief creates the runtime or returns the already existing one -> Singleton
     ///
     /// @param[in] name optional containing the name used for registering with the RouDi daemon
